dev/io.openliberty.netty.internal.tls.impl/src/io/openliberty/netty/internal/tls/impl/NettyTlsProviderImpl.java

```diff
@@ -0,0 +1,326 @@
+                }
+                throw e;
+            }
+        }
+
+        // at this point, we have a set of properties to use, which might be
+        // repertoire based or on-thread config... merge the channel props into
+        // them without overwriting any
+        if (null != props) {
+            Enumeration<?> names = properties.propertyNames();
+            String key = null;
+            String value = null;
+            while (names.hasMoreElements()) {
+                key = (String) names.nextElement();
+                value = properties.getProperty(key);
+                if (value instanceof String && null != value && !props.containsKey(key)) {
+                    props.put(key, value);
+                }
+            }
+        } else {
+            // otherwise we just use the channel config
+            props = properties;
+        }
+
+        // "SSSL" is a zOS repertoire type that is not supported by SSLChannel
+        // We only support "JSSE"
+        String sslType = (String) props.get(Constants.SSLPROP_SSLTYPE);
+        if (null != sslType) {
+            if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+                Tr.debug(tc, "SSLConfig type: " + sslType);
+            }
+            if (sslType.equals("SSSL")) {
+                throw new Exception("Invalid SSLConfig type: " + sslType);
+            }
+        }
+
+        // if debug is enabled, print out the properties we're going to use
+        if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+            Tr.debug(tc, "SSL configuration <null value means non-string>:");
+            Enumeration<?> names = props.propertyNames();
+            while (names.hasMoreElements()) {
+                String key = (String) names.nextElement();
+                String value = props.getProperty(key);
+                if (-1 == key.toLowerCase().indexOf("password")) {
+                    Tr.debug(tc, "\t" + key + " = " + value);
+                } else {
+                    // for nicer debug... print 1 * per character in the password
+                    StringBuilder output = new StringBuilder(4 + key.length() + value.length());
+                    output.append("\t").append(key).append(" = ");
+                    for (int i = 0; i < value.length(); i++) {
+                        output.append("*");
+                    }
+                    Tr.debug(tc, output.toString());
+                }
+            }
+        }
+
+        SSLContext context = null;
+        try {
+            SSLConfig config = new SSLConfig(props);
+            context = com.ibm.websphere.ssl.JSSEHelper.getInstance().getSSLContext(connectionInfo, config);
+        } catch (Exception e) {
+            // no FFDC required
+            if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+                Tr.debug(tc, "Exception getting SSLContext from properties.", new Object[] { e });
```

Comment:
should not be jsut debug and should include some of the key variables?

Reply:
I agree it could probably use some additional details. This is based off the SSL creation code in com.ibm.ws.channel.ssl [here](https://github.com/OpenLiberty/open-liberty/blob/e09e937bd8e87ee850c82fc5b6f8a50a736edb28/dev/com.ibm.ws.channel.ssl/src/com/ibm/ws/channel/ssl/internal/SSLChannel.java#L483) used also by channelfw. Not sure how closely related we want both of these to be so I created https://github.com/OpenLiberty/open-liberty/issues/24838 to follow up on this a bit more